Ruby: Event Manager: Run markdownlint autofixer (#27994)
TheOdinProject/curriculum: ruby/files_and_serialization/project_event_manager.md

- Utilize the data to contact a remote service.
- Populate a template with user data.
- Manipulate [strings](https://docs.ruby-lang.org/en/3.2/String.html).
- Access [Google's Civic Information API](https://developers.google.com/civic-information/) through the [Google API Client Gem](https://github.com/google/google-api-ruby-client).
- Use [ERB](https://docs.ruby-lang.org/en/3.2/ERB.html) (Embedded Ruby) for templating.

<div class="lesson-note" markdown="1">

  This tutorial is open source. If you notice errors, typos, or have questions/suggestions, please [open an issue on GitHub](https://github.com/TheOdinProject/curriculum/blob/main/ruby/files_and_serialization/project_event_manager.md).

> </div>

Reviewer: To fix this lint error, we can place this lesson note in the Introduction section (where it fits better anyway IMO).

Sarah 33703
David 07306
lib/event_manager.rb:11:in `block in <main>': undefined method `length' for nil:NilClass (NoMethodError)
 from /Users/burtlo/.rvm/rubies/ruby-1.9.3-p374/lib/ruby/1.9.1/csv.rb:1792:in `each'

>  from lib/event_manager.rb:7:in `<main>'

Reviewer: Was the partial leading trim to 1 space intentional here?

We now have our list of attendees with their valid zip codes (at least for most
of them). Using their zip code and the

> [Google Civic Information API](https://developers.google.com/civic-information/)

Reviewer: Nit: `'s` to match the first instance of this href's link text

```suggestion
[Google's Civic Information API](https://developers.google.com/civic-information/)
```

#### Showing all legislators in a zip code

The gem comes equipped with some vague example documentation. The documentation is also

> available online with their [source code for `goggle-api-ruby-client`](https://github.com/google/google-api-ruby-client).

Reviewer: :goggles: 
```suggestion
available online with their [source code for `google-api-ruby-client`](https://github.com/google/google-api-ruby-client).
```

Instead of outputting each raw legislator we want to print only their first
name and last name. We will need to complete the following steps:

- For each zip code, iterate over the array of legislators.
- For each legislator, we want to find the representative's name.
- Add the name to a new collection of names.

> To do this, we can use the built-in [`Array#map` method](https://docs.ruby-lang.org/en/3.2/Array.html#method-i-map). It works just like `.each` but returns a new array of the data we want to include.

Reviewer: ```suggestion
To do this, we can use the built-in [Array#map method](https://docs.ruby-lang.org/en/3.2/Array.html#method-i-map). It works just like `.each` but returns a new array of the data we want to include.
```
Having links where we have only one of the words in a code block (or vice versa) often renders not very well (looks like two links at a glance):
![image](https://github.com/TheOdinProject/curriculum/assets/122839503/c43226bc-43f3-4473-8c46-5a3f0cca3676)

Compared to
![image](https://github.com/TheOdinProject/curriculum/assets/122839503/66d93bf2-bbcb-4719-af1e-b9a9d8cb7ed8)

Instead of including the template within our application, we will instead load
the template using the same File tools we used at the beginning of the exercise.

> - Create a file named 'form_letter.HTML' in the root of your project directory.

Reviewer: Built-in rule fixing this only because it's not an inline code block. Using backticks keeps it happy.

```suggestion
- Create a file named `form_letter.html` in the root of your project directory.
```

- Assign an ID for the attendee

The first column does not have a name like the other columns, so we fall back to using the index value.

- Create an output folder

> We make a directory named "output" if a directory named "output" does not already exist.

Reviewer: Nit: Should *probably* be backticks here?
```suggestion
We make a directory named `output` if a directory named `output` does not already exist.
```

results of the variable `meaning_of_life`. We send the `result` message to the
template with `binding`.

- What is `binding`?

> The [`Kernel#binding` method](https://docs.ruby-lang.org/en/3.2/Kernel.html#method-i-binding)
> returns a special object. This object is an instance of the
> [`Binding` class](https://docs.ruby-lang.org/en/3.2/Binding.html). An instance of binding knows

Reviewer: These make sense! Just removing the backticks for the same rendering reason as above.
```suggestion
The [Kernel#binding method](https://docs.ruby-lang.org/en/3.2/Kernel.html#method-i-binding)
returns a special object. This object is an instance of the
[Binding class](https://docs.ruby-lang.org/en/3.2/Binding.html). An instance of binding knows
```

#### Defining an ERB template

> To use ERB we need to update our current template **form_letter.HTML**.

Reviewer: ```suggestion
To use ERB we need to update our current template `form_letter.html`
```

> - Save a new template as **form_letter.erb**

Reviewer: ```suggestion
- Save a new template as `form_letter.erb`
```
